File: src_qt/rules/RootNode.cpp
#include <QtCore/QString>
#include <QtCore/QListIterator>

#include "RootNode.h"
#include "../Brain.h"

namespace CnotiMind
{

	RootNode::RootNode( Brain* brain, QObject* parent ):
		RuleNode( brain, parent)
	{
	}

	void RootNode::exec()
	{
		execChildren();
	}

	/*
		This method should not be called

		Unless in the future, variable are passed already to the root node.
	*/
	void RootNode::exec( QHash<QString, QString>& variables )
	{
		execChildren( variables );
	}

	/*
		Reimplementation. This is the only node that it is root
	*/
	bool RootNode::isRoot() const
	{
		return true;
	}

	/*
		Debug information
	*/
	QString RootNode::info( int depth) const
	{
		return "Root" + RuleNode::info( depth );
	}

	RootNode *RootNode::fromXML( const QString &qName, const QXmlAttributes &atts, Brain* brain, QObject *parent)
	{
		Q_UNUSED(atts);

		if( qName.compare( "Rules", Qt::CaseInsensitive ) == 0 )
		{
			return new RootNode(brain, parent);
		}
		return NULL;
	}
}

File: src_ios/CnotiMind/rules/RootNode.m
//
//  RootNode.m
//  CnotiMind
//

#import "RootNode.h"

@implementation RootNode

- (id) initWithBrainAndParent:(Brain*)aBrain 
                       parent:(id)aParent
{
    if (self == [super initWithBrainAndParent:aBrain parent:aParent]) {
        
    }
    
    return self;
}

- (void) exec
{
    [self execChildren];
}

/*
 This method should not be called
 Unless in the future, variable are passed already to the root node.
 */
- (void) exec:(NSMutableDictionary*)aVariables
{
    [self execChildren:aVariables];
}

- (BOOL) isRoot
{
    return TRUE;
}

- (NSString*) info:(int)aDepth
{
    NSString* info = [NSString stringWithFormat:@""];
    info = [info stringByAppendingFormat:@"Root%@", [super info:aDepth]];
    
    return info;
}

@end
